tensorflow_addons/metrics/f1_scores.py

```diff
@@ -0,0 +1,166 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Implements F1 scores."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow.keras.metrics import Metric
+import numpy as np
+
+
+class F1_micro(Metric):
+    """
+    Calculates F1 micro score
+    """
+    def __init__(self, name='f1-score'):
```

Comment:
Nit: f1-score -> f1_score

Reply:
I think `F1Micro` should be a better Python class name.

BTW, my thought on different average scheme for f1 score:
```python
class F1Score(Metric):
    def __init__(self, name="f1_score", average="micro"):
        # average could be one of "micro", "macro" or "weighted"
```
This would be much close to sklearn API.
https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html

I'm also fine with seperated `F1Micro`, `F1Macro` and `F1Weighted` though. What do you free about this @Squadrick.